supply/lib/supply/uploader.rb

```diff
@@ -105,13 +117,23 @@ def upload_binary
             upload_changelog(language, apk_version_code)
           end
         end
-
       else
         Helper.log.info "No apk file found, you can pass the path to your apk using the `apk` option"
       end
+
+      UI.user_error!("Could not upload #{apk_path}") unless apk_version_code
```

Comment:
I think this line needs to move up inside the `if apk_path` branch. My test setup didn't have an apk path param (but also didn't have skip_upload_apk) and the behavior from the previous version of the tool changes such that I get the "Could not upload" error. Your code here might be more correct but let's not introduce a behavior change